### PR TITLE
chore(ide-extension): upgrade project setup 

### PR DESCRIPTION
**Pull request by openscript**
- Adds a dev container configuration
- Updates contributing.md according to the dev container configuration
- Updates dependencies
- Enhances project configuration
- Enhances launch scripts
- Removes Rollup (VSCode extension template removed it too. An option is to use webpack, but it makes the project setup complicated and I don't think inlang needs it)

### CODE REVIEW

**Review by samuelstroschein — CHANGES_REQUESTED** [review]
I propose to not merge this PR due to: 

a) The PR introduces changes that are unrelated to the container setup (and might be reverted by the follow up PR).
b) The container setup does not work for the codebase except for the ide extension.
c) Merging this PR will likely lead to merge conflicts. Numerous changes are required that will lead to merge conflicts with the follow-up PR that re-introduces the IDE extension. I propose waiting for the other PR first. 

Regarding b), the container setup only works for the ide extension. Turborepo, and therefore the entire code base, fails to run. I suspect the "node-typescript" dev container is to blame because it only includes "node"? 

I take that https://github.com/inlang/inlang/issues/312 is a larger undertaking but dev containers seem to be the way to go and I should set them up!

If you agree with closing this PR, wait for the follow-up PR, and then tackle the dev setup, close this PR. Otherwise, let me know how to continue

File: source-code/ide-extension/rollup.config.js [hunk 1]
```diff
@@ -1,29 +0,0 @@
-import typescript from "@rollup/plugin-typescript";
-import { nodeResolve } from "@rollup/plugin-node-resolve";
-
-/**
- * What is rollup used for?
- *
- * Inlang packages are ES Modules. VSCode only supports CJS (CommonJS modules).
- * Rollup is used to compile the ESM code to CJS.
- */
-
```
[the remark this answers is not in the document]
Comment (openscript) [reply source-code/ide-extension/rollup.config.js:10]: I don't completely understand if VSCode extensions really need to be CJS modules.

At the other hand officially VS Code doesn't use Rollup anymore. They removed it from their templates and tutorials.

What I try here is to stick to their proposed set up.

File: source-code/ide-extension/src/test/suite/index.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+      if (err) {
+        return e(err);
+      }
+
+      // Add files to the test suite
+      files.forEach(f => mocha.addFile(path.resolve(testsRoot, f)));
+
+      try {
+        // Run the mocha test
+        mocha.run(failures => {
```
[the remark this answers is not in the document]
Comment (openscript) [reply source-code/ide-extension/src/test/suite/index.ts:25]: Yes, I think this is possible.

File: source-code/ide-extension/tsconfig.json [hunk 3]
```diff
@@ -1,16 +1,21 @@
...
+  ],
   "compilerOptions": {
-    "lib": ["ESNext"],
+    "module": "commonjs",
+    "target": "ES2020",
     "outDir": "dist",
-    // Not specifying node16 causes https://github.com/rollup/plugins/issues/1224#issuecomment-1242109375
-    // interestingly, extending a base config taht contains module node16 still requires explicit manual
-    // setting of module node16 here.
-    "module": "Node16",
```
[the remark this answers is not in the document]
Comment (openscript) [reply source-code/ide-extension/tsconfig.json:10]: Yes, I agree.

File: source-code/ide-extension/package.json [hunk 4]
```diff
@@ -10,41 +10,51 @@
...
+    "commands": [
+      {
+        "command": "inlang.helloWorld",
+        "title": "Hello World"
+      }
+    ]
+  },
+  "scripts": {
+    "vscode:prepublish": "npm run compile",
+    "compile": "tsc -p ./",
```
[the remark this answers is not in the document]
Comment (openscript) [reply source-code/ide-extension/package.json:40]: Yes, I agree.

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -7,18 +7,22 @@ href: /documentation/contributing
...
+   - Use the preconfigured launch scripts in Visual Studio Codes `Run and Debug` view.
+7. **Contribute** your changes via a upstream pull request.
 
-## Debugging
+### Workspaces
 
-1. (If running, stop `npm run dev`.)
-2. Press `F5`.
-3. Wait until the terminal is showing "listening on localhost:3000".
-4. Open "localhost:xxxx" in the browser that has been opened.
```
[the remark this answers is not in the document]
Comment (openscript) [reply CONTRIBUTING.md:24]: I'll add another paragraph for this and mention the launch configuration.

**Comment by openscript** [comment]
I could use turbo and successfully build the entire project. What errors you get?

**Comment by samuelstroschein** [comment]
> sh: 1: turbo: Exec format error

File: source-code/ide-extension/rollup.config.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (samuelstroschein) [reply source-code/ide-extension/rollup.config.js:10]: If it works great, one dependency less. Electron does not support ESM though. I suspect the official template to use CJS.

https://github.com/microsoft/vscode/issues/130367
https://github.com/standard-things/esm/issues/915

**Comment by openscript** [comment]
No problem for me if you open another one and I'm also happy to contribute there.

**Comment by samuelstroschein** [comment]
Closed by #363.

**Pull request closed without merging**